### PR TITLE
[moos-core]Fix build failure on Windows

### PR DESCRIPTION
**Pull request by JackBoosY**
Steps:
1. Add compile macros and includes
2. Fix empty directories in static.

See #5594

### CODE REVIEW

**Comment by JackBoosY** [comment]
Hi @pmnewman, can you help me see if I am correct?
Thanks.

File: ports/moos-core/portfile.cmake [hunk 1]
```diff
@@ -24,26 +21,31 @@ vcpkg_configure_cmake(
     PREFER_NINJA
     OPTIONS
         -DBUILD_SHARED_LIBS=${BUILD_SHARED}
-        -DCMAKE_ENABLE_EXPORT=OFF
+        -DCMAKE_ENABLE_EXPORT=ON
+        -DCMAKE_WINDOWS_EXPORT_ALL_SYMBOLS=ON
```
Comment (grdowns) [inline ports/moos-core/portfile.cmake]: Unfortunately we cannot use `CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS`. See [#5784 (comment)](https://github.com/Microsoft/vcpkg/pull/5784#issuecomment-476908733). This port should be marked static only if it cannot support dynamic linking, and we can ensure that using `vcpkg_check_linkage(ONLY_STATIC_LIBRARY)`. See the PR #5937. Thanks @JackBoosY!

**Review by grdowns — CHANGES_REQUESTED** [review]

File: ports/moos-core/portfile.cmake (hunk 1 above)
Comment (JackBoosY) [reply ports/moos-core/portfile.cmake]: Thanks @grdowns!

**Review by grdowns — APPROVED** [review]

**Comment by grdowns** [comment]
Hey @JackBoosY, the changes look good. I cleaned up some of the tool copying as we don't build any binaries due to disabling the dynamic builds, and I removed the extra messaging and simplified the license installation. Thanks for the update!

**Comment by JackBoosY** [comment]
@grdowns thanks.

**Pull request merged**